### PR TITLE
Fixes VSTS Bug 699450: [Feedback] Error creating file -

### PR DESCRIPTION
**Pull request by mkrueger**
NullReferenceException: Object reference not set to an instance of an
object

https://devdiv.visualstudio.com/DevDiv/_workitems/edit/699450

### CODE REVIEW

**Comment by mkrueger** [comment]
@monojenkins backport release-7.7

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor/DocumentContext.cs [hunk 1]
```diff
@@ -186,5 +186,15 @@ internal virtual Task<IReadOnlyList<Editor.Projection.Projection>> GetPartialPro
 		internal virtual void UpdateDocumentId (Microsoft.CodeAnalysis.DocumentId newId)
 		{
 		}
+
+		#region IDisposable Support
+
+		public bool IsDisposed { get; private set; }
+
+		public virtual void Dispose ()
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor/DocumentContext.cs]: Add a `public void Dispose ()` and a `protected virtual void OnDispose (bool disposing)` so we are not limited by API in the future and we can add code before and after the dispose itself

**Comment by mkrueger** [comment]
No the bug is in the text editor extension itself - and that is the solution to that. If you've a better idea to fix that I'm open for ideas.

That's my solution to fix it.

**Comment by slluis** [comment]
@monojenkins backport release-7.6

**Comment by slluis** [comment]
@monojenkins backport release-7.7

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor/DocumentContext.cs (hunk 1 above)
Comment (Therzok) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor/DocumentContext.cs]: Can we have this changed before merging?

**Comment by Therzok** [comment]
@monojenkins backport release-7.7

**Comment by Therzok** [comment]
@monojenkins backport release-7.6

**Comment by Therzok** [comment]
Made the changes to Dispose method.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/Document.cs [hunk 2]
```diff
@@ -624,6 +626,7 @@ internal void DisposeDocument ()
 			parsedDocument = null;
 			views = null;
 			viewsRO = null;
+			base.Dispose ();
```
Comment (slluis) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/Document.cs]:
```suggestion
			base.OnDispose (disposing);
```

Comment (Therzok) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/Document.cs]: Yes.

**Comment by Therzok** [comment]
@monojenkins backport release-7.6

**Comment by Therzok** [comment]
@monojenkins backport release-7.7

**Pull request merged**